tick/preprocessing/longitudinal_features_lagger.py

```diff
@@ -0,0 +1,250 @@
+    >>> # output comes as a list of sparse matrices or 2D numpy arrays
+    >>> product_features.__class__
+    <class 'list'>
+    >>> [x.toarray() for x in product_features] 
+    [array([[ 0.,  0.,  0.,  1.,  0.,  0.,  0.,  0.,  0.],
+            [ 0.,  0.,  0.,  0.,  1.,  0.,  0.,  0.,  0.],
+            [ 0.,  0.,  0.,  0.,  0.,  1.,  1.,  0.,  0.]]),
+     array([[ 1.,  0.,  0.,  1.,  0.,  0.,  0.,  0.,  0.],
+            [ 0.,  1.,  0.,  0.,  1.,  0.,  1.,  0.,  0.],
+            [ 0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.]])]
+    """
+
+    _attrinfos = {
+        "n_lags": {"writable": False},
+        "_mapper": {"writable": False},
+        "_n_init_features": {"writable": False},
+        "_n_output_features": {"writable": False},
+        "_n_intervals": {"writable": False},
+        "_cpp_preprocessor": {"writable": False},
+        "_fitted": {"writable": False}
+    }
+
+    def __init__(self, n_lags=0, n_jobs=-1):
+        Base.__init__(self)
+        self.n_lags = n_lags
+        self.n_jobs = n_jobs
+        self._cpp_preprocessor = None
+        self._reset()
+
+    def _reset(self):
+        """Resets the object its initial construction state."""
+        self._set("_n_init_features", None)
+        self._set("_n_output_features", None)
+        self._set("_n_intervals", None)
+        self._set("_mapper", {})
+        self._set("_cpp_preprocessor", None)
+        self._set("_fitted", False)
+
+    @property
+    def mapper(self):
+        """Get the mapping between the feature lags and column indexes.
+
+        Returns
+        -------
+        output : `dict`
+            The column index - feature mapping.
+        """
+        if not self._fitted:
+            raise ValueError("cannot get mapper if object has not been fitted.")
+        return self._mapper.copy()
+
+    def fit(self, X, censoring):
+        """Fit the feature lagger using the features matrices list.
+
+        Parameters
+        ----------
+        X : list of np.ndarray or list of scipy.sparse.csr_matrix,
+            list of length n_samples, each element of the list of 
+            shape=(n_intervals, n_features)
+            The list of features matrices.
+
+            
+        censoring : `np.ndarray`, shape=(n_samples, 1), dtype="uint64"
+            The censoring data. This array should contain integers in 
+            [1, n_intervals]. If the value i is equal to n_intervals, then there
+            is no censoring for sample i. If censoring = c < n_intervals, then 
+            the observation of sample i is stopped at interval c, that is, the 
+            row c - 1 of the correponding matrix. The last n_intervals - c rows
+            are then set to 0.
+
+        Returns
+        -------
+        output : `LongitudinalFeaturesProduct`
+            The fitted current instance.
+        """
+        self._reset()
+        base_shape = X[0].shape
+        X = LFP._check_X_consistency(X, base_shape, "float64")
```

Comment:
check_X_consistency should be in a tick.preprocessing.utils script.
But we can leave this like that for now